### PR TITLE
[Lang] Fix pylint rule R1703

### PR DESCRIPTION
**Pull request by LJHG**
Related issue = #3151

### CODE REVIEW

File: python/taichi/lang/ast/checkers.py [hunk 1]
```diff
@@ -86,14 +86,11 @@ def generic_visit(self, node):
...
         return
-        if (isinstance(node.iter, ast.Call)
-                and isinstance(node.iter.func, ast.Attribute)
-                and isinstance(node.iter.func.value, ast.Name)
-                and node.iter.func.value.id == 'ti'
-                and node.iter.func.attr == 'static'):
-            is_static = True
-        else:
-            is_static = False
+        is_static = bool(isinstance(node.iter, ast.Call)
```
Comment (yihong0618) [inline python/taichi/lang/ast/checkers.py:89]: why add bool() here?
maybe use all?
```python
all([a, b, c])
```

File: python/taichi/lang/stmt_builder.py [hunk 2]
```diff
@@ -236,10 +236,7 @@ def build_If(ctx, node):
             node.test.func, ast.Attribute)
         if is_static_if:
             attr = node.test.func
-            if attr.attr == 'static':
-                is_static_if = True
-            else:
-                is_static_if = False
+            is_static_if = bool(attr.attr == 'static')
```
Comment (yihong0618) [inline python/taichi/lang/stmt_builder.py:239]: and we do need `bool`?

File: python/taichi/lang/ast/checkers.py (hunk 1 above)
Comment (LJHG) [reply python/taichi/lang/ast/checkers.py:89]: I use bool according to the log of pylint.
`R1703: The if statement can be replaced with 'var = bool(test)'`
Do we need change the code into the `all([a, b, c])` format?

Comment (LJHG) [reply python/taichi/lang/ast/checkers.py:89]: @ailzhang  Thanks. I will fix that.

**Review by ailzhang — APPROVED** [review]
Thanks! Welcome to taichi community!

**Comment by sjwsl** [comment]
Could you also add this pylint rule in CI to verify your work? See here for more details. Thanks!

**Review by ailzhang — CHANGES_REQUESTED** [review]
oops yea we need to enable this in CI as well. Requesting changes for to make sure we don't merge it accidentally. Thanks @sjwsl !

**Comment by LJHG** [comment]
@ailzhang I have enabled this in CI.

**Review by ailzhang — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**